### PR TITLE
network: add basic unit coverage for dns refresh

### PR DESCRIPTION
**Pull request by goaway**
Signed-off-by: Mike Schore <mike.schore@gmail.com>

### CODE REVIEW

File: test/common/network/configurator_test.cc [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+#include "gtest/gtest.h"
+#include "library/common/network/configurator.h"
+
+using testing::_;
+using testing::Return;
+
+namespace Envoy {
+namespace Network {
+
+class ConfiguratorTest : public testing::Test {
```
Comment (junr03) [inline test/common/network/configurator_test.cc:12]: Do we also want an integration test?

**Review by junr03 — COMMENTED** [review]
wonderful thank you so much!

File: test/common/network/configurator_test.cc (hunk 1 above)
Comment (goaway) [reply test/common/network/configurator_test.cc:12]: I have one, will push a separate branch in a bit.

**Review by junr03 — APPROVED** [review]
Talked offline. Integration test will come in a subsequent PR

**Pull request merged**